### PR TITLE
fix(route): 南方周末

### LINKED ISSUE #14716: 南方周末: wrong path
### 路由地址

```routes
/infzm/:id
```

### 完整路由地址

```fullroutes
/infzm/1
```

### 相关文档

https://docs.rsshub.app/routes/traditional-media#nan-fang-zhou-mo

### 预期是什么？

正常工作

### 实际发生了什么？

HTTP/2 404: Error Message: wrong path

### 部署

RSSHub 演示 (https://rsshub.app)

### 部署相关信息

_No response_

### 额外信息

```shell
NOLOG
```

### 这不是重复的 issue

- [X] 我已经搜索了 [现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by KarasuShin**
## Involved Issue / 该 PR 相关 Issue

Close #14716

## Example for the Proposed Route(s) / 路由地址示例

```routes
/infzm/1
/infzm/2
/infzm/3
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

### CODE REVIEW

File: lib/routes/infzm/index.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import cache from '@/utils/cache';
+
+export const route: Route = {
+    path: '/:id',
+    parameters: { id: '南方周末频道 id, 可在该频道的 URL 中找到（即 https://www.infzm.com/contents?term_id=:id)' },
+    categories: ['traditional-media'],
+    example: '/infzm/1',
+    radar: [
+        {
+            source: ['infzm.com/contents?term_id=:id'],
```
Comment (TonyRL) [inline lib/routes/infzm/index.ts:16]: https://docs.rsshub.app/joinus/new-radar#source
`source` cannot match against a URL search parameters

**Review by TonyRL — COMMENTED** [review]
Please remove the deprecated content:
- lib/routes-deprecated/infzm/news.js
- `router.get('/infzm/:id', lazyloadRouteHandler('./routes/infzm/news'));` in `lib/router.js`

File: lib/routes/infzm/index.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            source: ['infzm.com/contents'],
+            target: (_, url) => (url ? `/infzm/${url.match(/contents\?term_id=(\d*)/)?.[1]}` : ''),
+        },
+    ],
+    name: '频道',
+    maintainers: ['KarasuShin', 'ranpox', 'xyqfer'],
+    handler,
+    description: `
+    下面给出部分参考：
```
Comment (TonyRL) [inline lib/routes/infzm/index.ts:24]:
```suggestion
    description: `下面给出部分参考：

```

File: lib/routes/infzm/index.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    | 推荐 | 新闻 | 观点 | 文化 | 人物 | 影像 | 专题 | 生活 | 视频 |
+    | ---- | ---- | ---- | ---- | ---- | ---- | ---- | ---- | ---- |
+    | 1    | 2    | 3    | 4    | 7    | 8    | 6    | 5    | 131  |
+`,
```
Comment (TonyRL) [inline lib/routes/infzm/index.ts:28]:
```suggestion
    | 1    | 2    | 3    | 4    | 7    | 8    | 6    | 5    | 131  |`,
```

File: lib/routes/infzm/index.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                    url: link,
+                    headers: {
+                        Referer: link,
+                        Cookie: cookie || `passport_session=${Math.floor(Math.random() * 100)};`,
+                    },
+                });
+                const $ = load(response.data);
+                description = $('div.nfzm-content__content').html() ?? '';
+
+                cache.set(key, description);
```
Comment (TonyRL) [inline lib/routes/infzm/index.ts:69]: Please use [cache.tryGet](https://docs.rsshub.app/joinus/advanced/use-cache#api) instead.

Comment (KarasuShin) [reply lib/routes/infzm/index.ts]: fixed in https://github.com/DIYgod/RSSHub/pull/14886/commits/f401296c8df1cf5d34c9ec57881c7a865feb9e52, thanks for your review.

**Pull request merged**